vespalib/src/vespa/vespalib/stllike/allocator.h

```diff
@@ -0,0 +1,28 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+namespace vespalib {
+
+/**
+ * std compliant allocator that will use a smart allocator
+ * that uses mmap prefering huge pages for large allocations.
+ * This is a good fit for use with std::vector and std::deque.
+ */
+template <typename T>
+class allocator_large {
+    using PtrAndSize = alloc::MemoryAllocator::PtrAndSize;
+public:
+    allocator_large() : _allocator(alloc::MemoryAllocator::select_allocator()) {}
+    using value_type = T;
+    T * allocate(std::size_t n) {
+        return static_cast<T *>(_allocator->alloc(n*sizeof(T)).first);
+    }
+    void deallocate(T * p, std::size_t n) {
+        _allocator->free(p, n*sizeof(T));
+    }
```

Comment:
Since this is technically a stateful allocator whose ability to free an allocated pointer depends on the `allocator` instance the pointer was originally returned from, I think it might be required to add `operator==` here for testing allocator equivalence. See https://en.cppreference.com/w/cpp/memory/allocator/operator_cmp and https://howardhinnant.github.io/allocator_boilerplate.html#new

Reply:
Fixed